constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/motion/widget/ViewTransition.java

```diff
@@ -100,6 +151,18 @@
     private int mClearsTag = UNSET;
     private int mIfTagSet = UNSET;
     private int mIfTagNotSet = UNSET;
+    private int mSharedValue = UNSET;
+    private int mSharedValueID = UNSET;
```

Comment:
before those variables maybe add something like:
// shared value management. mSharedValueId is the key we are watching,
// mSharedValueCurrent the current value for that key, and mSharedValueTarget
// is the target we are waiting for to trigger.